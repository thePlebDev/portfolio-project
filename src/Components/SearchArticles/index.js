import React from 'react';

import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Container = styled.div`
  text-align:center;
  margin-bottom:20px;
`
const Title = styled.div`
  font-size:1.3rem;
  color:rgb(30, 130, 87);
  cursor:pointer;
`

const Description = styled.div`
  color:rgba(0, 0, 0, 0.75);
`
const Filter = styled.div`
  color:rgb(30, 130, 87);
  font-size:0.85rem;
  margin-bottom:10px;
  margin-top:10px;
  cursor:pointer;
`

const SearchArticles =({title,description,filters,id})=>{

  return(
    <Container >
      <Link to={`/blog/${id}`}>
        <Title>
          {title}
        </Title>
      </Link>
        <Description>
          {description}
        </Description>
    </Container>
  )
}

export default SearchArticles
