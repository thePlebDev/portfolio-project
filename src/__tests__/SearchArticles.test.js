import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom'

import SearchArticles from '../Components/SearchArticles'



//1) take a snapshot
//2) check that important stuff renders
//3) check default props
//4) check custom props
//5) check actions

const findByAttr =(component,dataTestId)=>{
  const data = component.find(`[data-testid='${dataTestId}']`)
  return data
}

describe('testing the SearchArticles component',()=>{

  it('should test the snapshot',()=>{
    const tree = renderer.create(<Router><SearchArticles/></Router>).toJSON();
    expect(tree).toMatchSnapshot()
  })

  it('should render the important stuff',()=>{
    const wrapper = shallow(<SearchArticles/>)
    expect(findByAttr(wrapper,'title').length).toBe(1)
    expect(findByAttr(wrapper,'description').length).toBe(1)
  })

  it('should render the default props',()=>{
    const wrapper = shallow(<SearchArticles/>)
    expect(findByAttr(wrapper,'title').props().children).toBe('Enter a title prop')
    expect(findByAttr(wrapper,'description').props().children).toBe('Enter a description prop')
  })

  it('should render the custom props ',()=>{
    const wrapper = shallow(<SearchArticles title={'title'} description={'description'}/>)
    expect(findByAttr(wrapper,'title').props().children).toBe('title')
    expect(findByAttr(wrapper,'description').props().children).toBe('description')

  })
})
