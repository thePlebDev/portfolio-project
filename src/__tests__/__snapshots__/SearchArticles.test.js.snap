// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the SearchArticles component should test the snapshot 1`] = `
<div
  className="sc-bdnylx cmxPiG"
>
  <a
    href="/blog/Enter id prop"
    onClick={[Function]}
  >
    <div
      className="sc-gtssRu cCDtkl"
      data-testid="title"
    >
      Enter a title prop
    </div>
  </a>
  <div
    className="sc-dlnjPT dukkMK"
    data-testid="description"
  >
    Enter a description prop
  </div>
</div>
`;
